homeassistant/components/rest_command/__init__.py

```diff
@@ -131,25 +127,21 @@ async def async_service_handler(service: ServiceCall) -> ServiceResponse:
                 variables=service.data, parse_result=False
             )
 
-            headers = None
-            if template_headers:
-                headers = {}
-                for header_name, template_header in template_headers.items():
-                    headers[header_name] = template_header.async_render(
-                        variables=service.data, parse_result=False
-                    )
+            headers = {}
+            for header_name, template_header in template_headers.items():
+                headers[header_name] = template_header.async_render(
+                    variables=service.data, parse_result=False
+                )
 
             if content_type:
-                if headers is None:
-                    headers = {}
                 headers[hdrs.CONTENT_TYPE] = content_type
 
             try:
                 async with getattr(websession, method)(
                     request_url,
                     data=payload,
                     auth=auth,
-                    headers=headers,
+                    headers=headers or None,
```

Comment:
Is the `or None` really needed?

The headers are checked here https://github.com/aio-libs/aiohttp/blob/5e44ba465c2168c6e1842df441828ec5759093e1/aiohttp/client.py#L924-L938, and it doesn't seem to matter if None or an empty dict are passed

Reply:
Looking at aiohttp's current implementation sending an empty dict would indeed be fine. However explicitly passing `None` ensures we're not implictly depending on the underlying handling of the methods parameters which might change. The signature states a default `None` if no headers are supplied so it seemed the safest choice. 

(Also kept `None` to avoid changing unit tests)

Reply:
I don't quite buy the argument that aiohttp would suddenly blow up when passing empty headers, but let's keep it this way.